### ISSUE TITLE
Release 0.8.0

### ISSUE THREAD

**Issue by johannes-b** [keptn/keptn#3256]
# Keptn 0.8.0

* **Link to milestone(s): [Milestone 0.8.0](https://github.com/keptn/keptn/milestone/17)** 
* **Release Date: <= March 2th, 2021**
* **Release Manager: Firstname Lastname (Insert GitHub Handle here)** (**TODO: Specify a release manager**)

### Step 1 - Release of keptn/spec [OPTIONAL]

**In case no new keptn/spec is released:**
- [ ] This version of Keptn does not require a new version of keptn/spec. 

**else:**

* **Artifact:** GitHub Release `keptn/spec:0.2.0` (also referred as ``spec.keptn.sh/0.2.0``)
* **Assignee**: Firstname Lastname (insert GitHub handle here) (**TODO: Assign this step to someone**)
* **Repo**: https://github.com/keptn/spec
* **Note**: Not every Keptn release requires a new Keptn spec.

**Tasks:**
- [x] All release-related PRs/discussions are resolved and are merged into master (please check back with the Issue/PR owner)
- [x] keptn/go-utils is in sync with spec/cloudevents (e.g., using `keptn generate cloud-events`)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] *Draft* a new release 0.2.0 https://github.com/keptn/spec/releases with release notes (collect them from PR/discussion titles) on master branch - again, ensure a semantic version is used!
- [x] Ask for feedback about release notes via Slack (link to the draft release can only be seen by users with write access to the repo)
- [x] If you're happy with everything and the release notes have been approved/accepted, pull the trigger and release the spec
- [ ] Notify Developers on Slack that a new release of keptn/spec has been created and link it (**AUTOMATION WANTED**)

---
### (FB) Step 2 - Release of keptn/examples

* **Artifact:** GitHub Release `keptn/examples:0.8.0`
* **Assignee**: Firstname Lastname (insert GitHub handle here) (**TODO: Assign this step to someone**)
* **Repo**: https://github.com/keptn/examples
* **Note**: Use the same version as for keptn/keptn (every Keptn release requires a new release of Keptn examples)

- [x] All release-related feature/bugfix/hotfix branches are merged into master (please check back with the PR owner)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Created `release-0.8.0` branch: `git checkout -b release-0.8.0` - ensure semantic version is used
- [x] _on release-0.8.0 branch:_ releasenotes_develop.md renamed to releasenotes_**V0.8.0**.md (**AUTOMATION WANTED**)
- [x] _on release-0.8.0 branch:_ collect release-notes either based on GitHub Issues for this Release or via commit messages (**AUTOMATION WANTED**)
- [x] _on release-0.8.0 branch:_ Update README.md with links to *Maintained Examples and Keptn Use cases* (not necessary for pre-releases)
- [x] _on release-0.8.0 branch:_ Don't forget: `git commit`, `git push [-u origin release-0.8.0]`
- [x] Create PR from release-0.8.0 to master and request reviews (review of release notes is mandatory)
- [x] If the PR is mergeable and approved, get it merged into master (**don't squash merge**) 
- [x] *Draft* a new release https://github.com/keptn/examples/releases with the release notes on the `release-0.8.0` branch (not on **master**) (**AUTOMATION WANTED**)
- [x] If you're happy with everything, pull the trigger and release the examples
- [ ] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)
- [ ] Created new and empty release notes (releasenotes_develop.md) file on master branch

--- 
### (BWA) Step 3 - Release of Keptn kubernetes-utils

* **Artifact:** GitHub Release of `keptn/kubernetes-utils:v0.8.0` 
* **Assignee**: Firstname Lastname (insert GitHub handle here) (**TODO: Assign this step to someone**)
* **Repo**: https://github.com/keptn/kubernetes-utils
* **Note:** The `v` in the version, which can be used in GoLang using `go get github.com/keptn/kubernetes-utils@v0.8.0`

- [x] All release-related feature/bugfix/hotfix branches are merged into master (please check back with the PR owner)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Created `release-0.8.0` branch: `git checkout -b release-0.8.0` - ensure semantic version is used
- [x] _on release-0.8.0 branch:_ releasenotes_develop.md renamed to releasenotes_**V0.8.0**.md
- [x] _on release-0.8.0 branch:_ collect release-notes either based on GitHub Issues for this Release or via commit messages (**AUTOMATION WANTED**)
- [x] _on release-0.8.0 branch:_ Don't forget: `git commit`, `git push [-u origin release-0.8.0]`
- [x] Create PR from release-0.8.0 to master and request reviews (review of release notes is mandatory)
- [x] If the PR is mergeable and approved, get it merged into master (**don't squash merge**) 
- [x] *Draft* a new release https://github.com/keptn/kubernetes-utils/releases with the release notes on the `release-0.8.0` branch (not on **master**) - please note the prefix `v` in the version name (**AUTOMATION WANTED**)
- [x] If you're happy with everything, pull the trigger and release keptn/kubernetes-utils
- [ ] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)
- [x] Created new and empty release notes (releasenotes_develop.md) file on master branch

--- 
### (FB) Step 4 - Release of Keptn go-utils

* **Artifact:** GitHub Release of `keptn/go-utils:v0.8.0` 
* **Assignee**: Firstname Lastname (insert GitHub handle here) (**TODO: Assign this step to someone**)
* **Repo**: https://github.com/keptn/go-utils
* **Note:** The `v` in the version, which can be used in GoLang using `go get github.com/keptn/go-utils@v0.8.0`

- [x] All release-related feature/bugfix/hotfix branches are merged into master (please check back with the PR owner)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Created `release-0.8.0` branch: `git checkout -b release-0.8.0`
- [x] _on release-0.8.0 branch:_ version bump (**Is this needed?**)
- [x] _on release-0.8.0 branch:_ releasenotes_develop.md renamed to releasenotes_**V0.8.0**.md
- [x] _on release-0.8.0 branch:_ collect release-notes either based on GitHub Issues for this Release or via commit messages (**AUTOMATION WANTED**)
- [x] _on release-0.8.0 branch:_ Don't forget: `git commit`, `git push [-u origin release-0.8.0]`
- [x] Create PR from release-0.8.0 to master and request reviews (review of release notes is mandatory)
- [x] If the PR is mergeable and approved, get it merged into master (**don't squash merge**) 
- [x] *Draft* a new release https://github.com/keptn/go-utils/releases with the release notes on the `release-0.8.0` branch (not on **master**) - please note the prefix `v` in the version name (**AUTOMATION WANTED**)
- [ ] If you're happy with everything, pull the trigger and release keptn/kubernetes-utils
- [ ] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)
- [ ] Created new and empty release notes (releasenotes_develop.md) file on master branch

---

## Prepare release of associated services

:warning: **WARNING/STOP/DANGERZONE**: Do not start with this section unless **all the above steps** have been completed successfully.

---
### Step 5 - Create Release Branch for keptn-contrib services

**Please update version tags of below services accordingly; use semantic versioning**

**foreach:**
- [x] dynatrace-service > **dynatrace-service:x.x.x**  (BWA)
- [x] dynatrace-sli-service > **dynatrace-sli-service:x.x.x** (FB)
- [x] prometheus-service > **prometheus-service:x.x.x** (BWA)
- [x] prometheus-sli-service > **prometheus-sli-service:x.x.x** (BWA)
- [ ] argo-service > **argo-service:x.x.x** (no release yet)
- [ ] servicenow-service > **servicenow-service:x.x.x** (no release yet)
- [ ] notification-service > **notification-service:x.x.x** (no release yet)
- [x] unleash-service > **unleash-service:x.x.x** (BWA)

follow https://github.com/keptn/keptn/wiki/Keptn-Contrib-Release-Checklist Step 1

## Prepare Release of Keptn

---
### Step 6 - Release of Keptn
- [x] `keptn upgrade` - verify if any changes/migrations/... in the CLI command are needed for this release
- [x] `keptn upgrade` - Create a PR at https://github.com/keptn/get.keptn.sh and modify [version.json](https://github.com/keptn/get.keptn.sh/blob/master/src/version.json) accordingly
- [x] Upgrade docs available in https://keptn.sh/docs/
- [x] Git submodule for *specification* directory in keptn release branch are updated (e.g., `cd specification && git fetch && git checkout 0.2.0 && cd .. && git add specification && git commit -m "updated spec to 0.2.0" && git push`) 
   - **Note**: In case the `specification` directory in the keptn/keptn repo is empty, execute the following command **first**: `git submodule update --init`.
- [x] All relevant feature/bugfix/hotfix branches are merged into master
- [x] Update references to keptn-contrib services in [integration tests](https://github.com/keptn/keptn/blob/master/.github/workflows/integration_tests.yml) to the release branches created in Step 5. Currently used: unleash-service, dynatrace-sli-service
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Latest go-utils version is used (run: `go get github.com/keptn/go-utils@v0.8.0`) in all sub-folders that use it (see [this code snippet](https://github.com/keptn/go-utils/blob/247b98ef868c6234cf5cb5f852551c4746d472ea/.travis.yml#L56-L69) to use an example)
- [x] Latest kubernetes-utils version is used (run: `go get github.com/keptn/kubernetes-utils@v0.8.0`) in all sub-folders that use it: `api`, `cli`, `configuration-service`, `gatekeeper-service`, `helm-service`.
- [x] Keptn release notes are up-to-date and spec version is referenced in release notes 
- [x] releasenotes_develop.md renamed to releasenotes_**V0.8.0**.md (automated using "Create Release Branch Workflow")
- [x] Created `release-0.8.0` branch: `git checkout -b release-0.8.0` (automated using "Create Release Branch Workflow")
- [ ] Ask stakeholders to review the release notes (*Note:* Do not use a GitHub DRAFT Release, it does not allow editing from multiple parties at the same time and we will lose changes**)
- [x] Check [GH Workflow CI for release branch](https://github.com/keptn/keptn/actions?query=workflow%3ACI+branch%3Arelease-0.8.0)
- [x] Check [GH Workflow Integration Test for release branch](https://github.com/keptn/keptn/actions?query=workflow%3A%22Integration+Tests%22+branch%3Arelease-0.8.0)
- [x] Check for a draft release with the correct version https://github.com/keptn/keptn/releases (**DO NOT RELEASE IT YET**)
- [x] Draft release contains CLI binaries for Linux, Windows and MacOS
- [x] Draft release contains Keptn installer helm chart
- [x] Draft release contains Helm Charts for jmeter-service and helm-service
- [x] Create a *draft* PR from release-0.8.0 to master (automated using "Create Release Branch Workflow")

Check on GCP Cloud Storage (Attention: This is not automated at the moment):
- [x] keptn-installer 0.8.0 (also index file needs to be updated)

Notify certain users and stakeholders about the release, such that they can test it and ensure compatibility with their services/docs. Make sure to provide instructions on how to download the CLI and install Keptn. **Make sure to let them know that this might not the final release, rather that it is a release candidate.**

- [ ] Notify developers, keptn/tutorials CODEOWNERS, keptn-contrib and keptn-sandbox repo owners via Keptn Slack (e.g., keptn-integrations channel) about the upcoming release and provide a preview to the release notes

---
### Step 7 - Final Preparation phase

- [x] keptn-sockshop/carts released 
- [x] keptn.sh/docs/integrations/ is up-to-date
- [x] keptn CLI command docu generated and updated on keptn.sh/docs/x.x.x/reference/cli/commands/ (**TODO: Update link to docs**)
- [x] keptn.sh/docs/ is up-to-date
- [x] tutorials.keptn.sh: update of tutorials in this branch/PR: xyz (**TODO: Add link to PR/branch**)

---
### Step 8 - Test phase

Upgrader:
- [ ] Upgrade command tested for full installation (GKE or K3s)
- [ ] Upgrade command tested for quality-gates/control-plane installation (GKE or K3s)
- [ ] Upgrade command tested for full installation (OpenShift 3.11 or MiniShift)
- [ ] Upgrade command tested for quality-gates/control-plane installation (OpenShift 3.11 or MiniShift)

Tests:
- Download of **keptn CLI 0.8.0** > [ToDo: LINK to CLI]()
- Install keptn cli and verify that it prints the correct version: `keptn version`
- For keptn-contrib services: follow https://github.com/keptn/keptn/wiki/Keptn-Contrib-Release-Checklist Step 2
- Pre-view of **keptn.sh/docs** > [ToDo: LINK]()

**Note**: The following is partially covered by integration tests. For pre-releases it can be skipped.
However, for GA releases the tests in this checklist need to be executed manually.

- [ ] GCP - GKE - **Keptn core team**
  - [x] installation: `keptn install`
    - [ ] quality gates UC completed according to tutorials.keptn.sh with Dynatrace
    - [ ] quality gates UC completed according to tutorials.keptn.sh with Prometheus
  - [x] Installation: `keptn install --use-case=continuous-delivery`
    - [ ] All tutorials completed with Dynatrace
        - [x] Onboarding a Service
        - [x] Deployments with Quality Gates
        - [ ] Up-scale with Dynatrace
        - [x] Self-healing with Feature Flags
    - [ ] All tutorials completed with Prometheus 
        - [ ] Onboarding a Service
        - [ ] Deployments with Quality Gates
        - [ ] Up-scale with Prometheus
  - [x] Bridge verified
  - [ ] Delete a project used in the use cases: `keptn delete project`
  - [x] Uninstall Keptn from cluster: `keptn uninstall`
  - [x] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used K8s version
- [x] Azure - AKS @christian-kreuzberger-dtx 
  - [x] Installation: `keptn install --use-case=continuous-delivery`
  - [x] Ensure `keptn version` prints the correct version
  - [x] `keptn create project xyz`
  - [x] Ensure Keptn Bridge can be accessed and works
  - [x] Uninstall Keptn from cluster: `keptn uninstall`
  - [x] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used K8s version
- [ ] AWS - EKS
  - [ ] All tutorials completed with Dynatrace
      - [ ] Onboarding a Service
      - [ ] Deployments with Quality Gates
      - [ ] Up-scale with Dynatrace
      - [ ] Self-healing with Feature Flags
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used K8s version
- [ ] (AG) OpenShift 3.11 - MiniShift
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used OpenShift version
- [ ] OpenShift 4.x
  - [ ] All tutorials completed with Dynatrace
      - [ ] Onboarding a Service
      - [ ] Deployments with Quality Gates
      - [ ] Self-healing with Feature Flags
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used OpenShift version
  - [ ] Update support matrix on keptn.sh/docs and release notes (key announcement + special thanks to Marc)
- [ ] Kubernetes (e.g., K3s, Rancher)
  - [x] Installation: `keptn install --use-case=continuous-delivery`
  - [x] Ensure `keptn version` prints the correct version
  - [ ] `keptn create project xyz`
  - [ ] Ensure Keptn Bridge can be accessed and works
  - [ ] Uninstall Keptn from cluster: `keptn uninstall`
  - [x] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used K8s version

---
### Step 9 - Official release of keptn-contrib services

**foreach:**
- [x] dynatrace-service > **dynatrace-service:x.x.x** (JB)
- [x] dynatrace-sli-service > **dynatrace-sli-service:x.x.x** (JB)
- [x] prometheus-service > **prometheus-service:x.x.x** (JB)
- [x] prometheus-sli-service > **prometheus-sli-service:x.x.x** (JB)
- [ ] argo-service > **argo-service:x.x.x** (no release yet)
- [ ] servicenow-service > **servicenow-service::x.x.x** (no release yet)
- [ ] notification-service > **notification-service:x.x.x** (no release yet)
- [x] unleash-service > **unleash-service:x.x.x** (CKR)

follow https://github.com/keptn/keptn/wiki/Keptn-Contrib-Release-Checklist Step 3

---
### Step 10 - Official release of Keptn

**Note**: Some steps are only for GA releases, not for alpha, beta, etc.
- [x] Merge the PR at https://github.com/keptn/get.keptn.sh [version.json](https://github.com/keptn/get.keptn.sh/blob/master/src/version.json)

Check on GCP Cloud Storage (Attention: This is not automated anymore): 
- [ ] keptn-cli x.x.x
- [x] keptn-installer x.x.x

Release Keptn
- [x] Publish the draft release for Keptn 0.8.0 on GitHub, select `release-0.8.0` as a target
- [x] Verify that installing Keptn CLI works via `curl -sL https://get.keptn.sh | KEPTN_VERSION=0.8.0 bash` and execute `keptn version` to check the version number
- [x] Merge PR from **release-x.x.x** to **master** (**merge only if needed**)
- [x] Created new and empty release notes (releasenotes_develop.md) file on master branch

---
### Final steps
- [x] Update **Quick Start** on keptn.sh with newest release (**only for GA, not for pre-releases**)
- [x] Update [get.keptn.sh repo](https://github.com/keptn/get.keptn.sh) (which auto-uploads to get.keptn.sh S3 bucket with) newest release (**only for GA, not for alpha/beta releases**)
- [x] Add release notes to: https://keptn.sh/docs/news/release_announcements/ (**only for GA, not for pre-releases; AUTOMATION WANTED**)
- [x] Send out Slack notification
- [ ] Send out Twitter notification
- [ ] Send out Release-Email
- [ ] Make sure any (relevant) changes to the release checklist are also saved in the [wiki](https://github.com/keptn/keptn/wiki/Release-Checklist)
